packages/cli-repl/src/cli-repl.ts

```diff
@@ -1206,11 +1207,14 @@ export class CliRepl implements MongoshIOProvider {
    * @param code The user-provided exit code, if any.
    */
   async exit(code?: number): Promise<never> {
+    this.loggingAndTelemetry?.flush();
```

Comment:
I think we should be doing this generally in any case to make sure all telemetry (including the error afterwards?) get reported even before this change